src/3box.js

```diff
@@ -117,30 +117,40 @@ class ThreeBox {
       // console.log(this._rootStore.iterator({ limit: -1 }).collect().length)
       // console.log(await this._ipfs.pubsub.peers('/orbitdb/QmRxUAGk62v7NjUkzvcqwYkBqF3zHb8tfhfW6T3MateGje/b932fe7ab.root'))
       console.time('replicate rootStore')
-      if (!this._rootStore.iterator({ limit: -1 }).collect().length) {
+      const entries = this._rootStore.iterator({ limit: -1 }).collect()
+      const rootStoreEmpty = !entries.length
+      // const rs = this._rootStore
+      // function replicatedRootStore () {
+      //   return new Promise((resolve, reject) => {
+      //     rs.events.on('replicated', () => {
+      //       console.timeEnd('replicate rootStore')
+      //       resolve()
+      //     })
+      //   })
+      // }
+      const dbSyncPromise = (entry) => {
+        const odbAddress = entry.payload.value.odbAddress
+        const name = odbAddress.split('.')[1]
+        if (name === 'public') return this.public._sync(odbAddress)
+        if (name === 'private') return this.private._sync(odbAddress)
+      }
+      let storePromises = []
+      console.time('sync stores')
+      if (rootStoreEmpty) {
+        console.log('root store empty ')
+        // storePromises.push(replicatedRootStore())  // would have thought this did that same as wrapping below in promise??
         await new Promise((resolve, reject) => {
-          this._rootStore.events.on(
-            'replicate.progress',
-            (_x, _y, _z, num, max) => {
-              if (num === max) {
-                console.timeEnd('replicate rootStore')
-                this._rootStore.events.on('replicated', resolve)
-              }
+          this._rootStore.events.on('replicate.progress', (address, hash, entry, progress, have) => {
```

Comment:
The `'replicate.progress'` event is emitted as soon as an event has been received from the network, but not added to the local odb yet.